openedx/adg/lms/applications/helpers.py

```diff
@@ -365,67 +365,79 @@ def get_course_card_information(user, courses):
             )
 
-        course_cards_information.append({
-            'course': course,
-            'status': status,
-            'grade': grade,
-            'message': message
-        })
+        if status != COMPLETED:
+            are_courses_completed = False
 
-    return course_cards_information, is_any_course_started, is_locked
+        if pre_requisite_courses:
+            if status == LOCKED:
+                locked_course_cards_information.append({
+                    'course': course, 'status': status, 'grade': grade, 'message': message, 'pre_req': pre_req
+                })
+            else:
+                unlocked_course_cards_information.append({
+                    'course': course, 'status': status, 'grade': grade, 'message': message, 'pre_req': pre_req
+                })
+        else:
+            course_cards_information.append({
+                'course': course, 'status': status, 'grade': grade, 'message': message, 'pre_req': pre_req
+            })
+
+    course_cards = course_cards_information + unlocked_course_cards_information + locked_course_cards_information
+    return course_cards, is_any_course_started, are_courses_completed
 
 
-def get_information_for_course_with_prerequisites(user, pre_requisite_courses, is_locked, status):
+def get_information_for_course_with_prerequisites(user, pre_requisite_courses, status):
     """
-    Checks if all prerequisite courses have been passed, if not then changes the status to Locked, sets the is_locked
-    flag to True and returns a message advising the user to pass the prerequisite courses.
+    Checks if all prerequisite courses have been passed, if not then changes the status to Locked and returns a message
+    advising the user to pass the prerequisite courses.
 
     Arguments:
         user (User): User object
         pre_requisite_courses (list): List containing prerequisite courses
-        is_locked (bool): Flag to indicate if any course is locked
         status (str): Contains status of a course
 
     Returns:
-        is_locked (bool): Flag to indicate if any course is locked
         status (str): Contains status of a course
         message (str): Message to indicate the user to pass the prerequisite courses
+        prerequisite_name (str): Name of the prerequisite course
     """
     message = LOCKED_COURSE_MESSAGE
+    prerequisite_name = ''
 
     for prereq in pre_requisite_courses:
         course_grade = CourseGradeFactory().read(user, course_key=prereq['key'])
+        course_overview = CourseOverview.objects.get(id=prereq['key'])
+        prerequisite_name = course_overview.display_name_with_default
 
         if not (course_grade and course_grade.passed):
-            is_locked = True
             status = LOCKED
-
-            course_overview = CourseOverview.objects.get(id=prereq['key'])
-            message = f'{message} {course_overview.display_name_with_default},'
+            message = f'{message} {prerequisite_name},'
```

Comment:
This function gets a list of prerequisite courses and in this loop we are appending the prerequisite course names to the end of the message. 

While redefining the message, I have also included message inside the f-string, meaning that this will append the prerequisite course name at the end of the message. Hence every time we reach this part of code a new course name will be added.

Reply:
As for second question we can use index like functionality for stings as well.

Reply:
Got it, but there will be only one `prerequisite_name` returned by the function via:

`return status, message, prerequisite_name`

^ here `prerequisite_name` will only contain the name of the last prereq course in `pre_requisite_courses` list, right?